include/seqan3/alignment/pairwise/detail/policy_affine_gap_with_trace_recursion.hpp

```diff
@@ -78,9 +78,11 @@ class policy_affine_gap_with_trace_recursion : protected policy_affine_gap_recur
         diagonal_score = (diagonal_score < vertical_score)
                            ? (best_trace = previous_cell.vertical_trace(), vertical_score)
                            : (best_trace |= previous_cell.vertical_trace(), diagonal_score);
-        diagonal_score = (diagonal_score < horizontal_score)
-                           ? (best_trace = previous_cell.horizontal_trace(), horizontal_score)
-                           : (best_trace |= previous_cell.horizontal_trace(), diagonal_score);
+        diagonal_score =
+            (diagonal_score < horizontal_score)
+                ? (best_trace = previous_cell.horizontal_trace() | (best_trace & trace_directions::carry_up_open),
+                   horizontal_score)
+                : (best_trace |= previous_cell.horizontal_trace(), diagonal_score);
```

Comment:
Just a small remark. Can we write this as an if statement? It is really hard to read. ^^

Reply:
I also find it hard to read. However, the ternary operator (`<condition> ? <instructions if true> : <instructions if false>`) is usually better optimized by the compiler than an `if` branch. Since this code is performance-relevant, a ternary is more suited.

Reply:
@eseiler is right here. In fact the compiler can vectorise a ternary operator, while it seems to be more complicated with an if branch.